profile/profile.go

```diff
@@ -68,18 +68,27 @@ type Sample struct {
 	Location []*Location
 	Value    []int64
 	Label    map[string][]string
-	NumLabel map[string][]int64
+	NumLabel map[string][]NumValue
```

Comment:
Suggestion: this change technically breaks the API of Sample which is public. I don't know if this is going to break any existing clients, but could consider doing

```
NumLabel map[string][]int64
NumLabelUnit map[string][]string
```

instead. That is less memory efficient though so maybe save that only for the case there are clients we can't upgrade easily.